### PR TITLE
[10.0] Improve handling of refunds

### PR DESCRIPTION
**Pull request by alexis-via**
Ability to sort payment lines by maturity date

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@carlosdauden can you please review this as an alternative to #469?

**Review by carlosdauden — APPROVED** [review]
Code review

**Comment by chrisdec** [comment]
Hello @alexis-via, I made some functional tests but it seems it only works for supplier refunds and not for customer refunds. That's right? 

In V8, It was possible to add customer refunds to payment order. Do you know a way to do this in V10?

Thanks,

**Comment by jbaudoux** [comment]
@alexis-via @chrisdec I tested the customer refunds and it seems to work with 2 remarks:
-  The payment_mode is not set on the refunds. I suggest to merge this PR https://github.com/akretion/banking/pull/6
- There should be a way to remove payment lines having a negative total as this blocks the creation of the file

File: account_payment_order/wizard/account_payment_line_create.py [hunk 1]
```diff
@@ -93,29 +98,19 @@ def _prepare_move_line_domain(self):
...
-            domain += [
-                ('credit', '>', 0),
-                #  '|',
-                ('account_id.internal_type', '=', 'payable'),
-                #  '&',
-                #  ('account_id.internal_type', '=', 'receivable'),
-                #  ('reconcile_partial_id', '=', False),  # TODO uncomment
-            ]
+            domain += [('account_id.internal_type', '=', 'payable')]
+            if not self.refund:
```
Comment (jbaudoux) [inline account_payment_order/wizard/account_payment_line_create.py:102]:
```suggestion
            if not self.refund or not self.order_id.payment_mode_id.group_lines:
```

File: account_payment_order/wizard/account_payment_line_create.py [hunk 2]
```diff
@@ -93,29 +98,19 @@ def _prepare_move_line_domain(self):
...
+                # exclude lines with amount = 0, that may come from invoices
+                # with amount = 0
+                domain += ['|', ('credit', '>', 0), ('debit', '>', 0)]
         elif self.order_id.payment_type == 'inbound':
-            domain += [
-                ('debit', '>', 0),
-                ('account_id.internal_type', '=', 'receivable'),
-                ]
+            domain += [('account_id.internal_type', '=', 'receivable')]
+            if not self.refund:
```
Comment (jbaudoux) [inline account_payment_order/wizard/account_payment_line_create.py:110]:
```suggestion
            if not self.refund or not self.order_id.payment_mode_id.group_lines:
```

**Review by jbaudoux — COMMENTED** [review]
If payment lines are not grouped, you cannot include refunds. It will be blocked at file generation as you cannot have negative payment lines.

**Pull request closed without merging**